### ISSUE TITLE
witti error: Block has an unknown parent: 0xa243…8e13, id: 4, service: sync

### ISSUE THREAD

**Issue by q9f** [sigp/lighthouse#1272]
## Description

Lighthouse fails to sync Witti.

## Version

`master` @ 9db0c280510bb0d8ceaafe61f1e2f9ff39707ca5

## Present Behaviour

https://termbin.com/5zr6

```
Jun 16 14:07:05.220 WARN Batch processing failed                 error: Block has an unknown parent: 0xa243…8e13, id: 4, service: sync
Jun 16 14:07:05.448 WARN Batch processing failed                 error: Block has an unknown parent: 0xa243…8e13, id: 4, service: sync
Jun 16 14:07:06.000 INFO Syncing                                 est_time: 2 hrs 1 mins, speed: 18.67 slots/sec, distance: 135860 slots (2 weeks 4 days), peers: 12, service: slot_notifier
Jun 16 14:07:06.194 WARN Batch processing failed                 error: Block has an unknown parent: 0xa243…8e13, id: 4, service: sync
Jun 16 14:07:06.360 WARN Batch processing failed                 error: Block has an unknown parent: 0xa243…8e13, id: 4, service: sync
Jun 16 14:07:07.317 WARN Batch processing failed                 error: Block has an unknown parent: 0xa243…8e13, id: 4, service: sync
Jun 16 14:07:07.319 WARN Batch failed to download. Dropping chain and downvoting peers, id: 4, chain_id: 5626889166973438031, service: sync
Jun 16 14:07:07.320 WARN Disconnecting peer (RPC)                peer_id: PeerId("16Uiu2HAmJACYTUNi88cfHqJ6QSJPKZokwQ8WwA4YmE8FcUaUdtEw"), reason: Fault, service: sync
Jun 16 14:07:07.321 WARN Disconnecting peer (RPC)                peer_id: PeerId("16Uiu2HAkuaSHYRBfSsVrJVdbaiPcJ4AGHwDxHbhg6xiBDCesPVDt"), reason: Fault, service: sync
Jun 16 14:07:07.322 WARN Disconnecting peer (RPC)                peer_id: PeerId("16Uiu2HAmDPTSbDzkAPGwLTuGKH1vtcAPhE8iVEFkLWjJdz2NdkUQ"), reason: Fault, service: sync
Jun 16 14:07:07.323 WARN Disconnecting peer (RPC)                peer_id: PeerId("16Uiu2HAmTXz3fiywgagb1wLE3eFBFqYUjwgAgijetQyq7SayYf1L"), reason: Fault, service: sync
Jun 16 14:07:07.324 WARN Disconnecting peer (RPC)                peer_id: PeerId("16Uiu2HAmNRyPwJUaSAqwiLQA9k3ibZnTkqSHyQvYrtTBbDTgkdL5"), reason: Fault, service: sync
Jun 16 14:07:07.326 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAmJACYTUNi88cfHqJ6QSJPKZokwQ8WwA4YmE8FcUaUdtEw, service: libp2p
Jun 16 14:07:07.327 WARN Disconnecting peer (RPC)                peer_id: PeerId("16Uiu2HAkvfDTKtFoSU5CHoA47BRwcJhN2zfvmr1kZXAwbweV3pq2"), reason: Fault, service: sync
Jun 16 14:07:07.328 WARN Disconnecting peer (RPC)                peer_id: PeerId("16Uiu2HAmK3aw5p4Uw7RYRFeUcL4u1pg3u6JN8MnyT5wRshNLvHqU"), reason: Fault, service: sync
Jun 16 14:07:07.329 WARN Disconnecting peer (RPC)                peer_id: PeerId("16Uiu2HAmEM6HYXuMCeKUeMtmjYdQsz23MekUrp2TB3bADiijLj3e"), reason: Fault, service: sync
Jun 16 14:07:07.330 WARN Disconnecting peer (RPC)                peer_id: PeerId("16Uiu2HAkucMAVckq8yyLU51pbnzjTZ6JptPLKdpH5USBGuuvinHP"), reason: Fault, service: sync
Jun 16 14:07:07.331 WARN Disconnecting peer (RPC)                peer_id: PeerId("16Uiu2HAm295TBWmcnwmUTw3KJB3NQ7iGtWs2crUZvc1RKAPSZzLe"), reason: Fault, service: sync
Jun 16 14:07:07.332 WARN Disconnecting peer (RPC)                peer_id: PeerId("16Uiu2HAmHZoF5j6tByYmRBUT37WfTf8JpDThRThou6bUpv4TxzUj"), reason: Fault, service: sync
Jun 16 14:07:07.333 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAkuaSHYRBfSsVrJVdbaiPcJ4AGHwDxHbhg6xiBDCesPVDt, service: libp2p
Jun 16 14:07:07.334 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAmDPTSbDzkAPGwLTuGKH1vtcAPhE8iVEFkLWjJdz2NdkUQ, service: libp2p
Jun 16 14:07:07.335 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAmTXz3fiywgagb1wLE3eFBFqYUjwgAgijetQyq7SayYf1L, service: libp2p
Jun 16 14:07:07.336 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAmNRyPwJUaSAqwiLQA9k3ibZnTkqSHyQvYrtTBbDTgkdL5, service: libp2p
Jun 16 14:07:07.337 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAkvfDTKtFoSU5CHoA47BRwcJhN2zfvmr1kZXAwbweV3pq2, service: libp2p
Jun 16 14:07:07.338 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAmK3aw5p4Uw7RYRFeUcL4u1pg3u6JN8MnyT5wRshNLvHqU, service: libp2p
Jun 16 14:07:07.339 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAmEM6HYXuMCeKUeMtmjYdQsz23MekUrp2TB3bADiijLj3e, service: libp2p
Jun 16 14:07:07.340 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAkucMAVckq8yyLU51pbnzjTZ6JptPLKdpH5USBGuuvinHP, service: libp2p
Jun 16 14:07:07.341 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAm295TBWmcnwmUTw3KJB3NQ7iGtWs2crUZvc1RKAPSZzLe, service: libp2p
Jun 16 14:07:07.342 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAmHZoF5j6tByYmRBUT37WfTf8JpDThRThou6bUpv4TxzUj, service: libp2p
[2020-06-16T12:07:07Z ERROR yamux::connection] 09f06662: socket error: decode error: i/o error: I/O error: Connection reset by peer (os error 104)
[2020-06-16T12:07:07Z ERROR yamux::connection] a971e841: socket error: decode error: i/o error: I/O error: Connection reset by peer (os error 104)
[2020-06-16T12:07:07Z ERROR yamux::connection] 23578777: socket error: decode error: i/o error: I/O error: Connection reset by peer (os error 104)
Jun 16 14:07:18.001 INFO Syncing                                 est_time: 1 hrs 56 mins, speed: 19.33 slots/sec, distance: 135605 slots (2 weeks 4 days), peers: 10, service: slot_notifier
Jun 16 14:07:30.000 INFO Syncing                                 est_time: 2 hrs 21 mins, speed: 15.98 slots/sec, distance: 135543 slots (2 weeks 4 days), peers: 10, service: slot_notifier
Jun 16 14:07:42.000 INFO Syncing                                 est_time: 4 hrs 43 mins, speed: 7.98 slots/sec, distance: 135544 slots (2 weeks 4 days), peers: 10, service: slot_notifier
Jun 16 14:07:54.001 INFO Syncing                                 est_time: 5 hrs 39 mins, speed: 6.65 slots/sec, distance: 135545 slots (2 weeks 4 days), peers: 10, service: slot_notifier
Jun 16 14:07:59.187 WARN Disconnecting and banning peer          timeout: 30s, peer_id: 16Uiu2HAmCVbB3Mp2H4sAao5mNRggPiQAPjyLqyzmovLBeA6K8x2C, service: libp2p
[2020-06-16T12:07:59Z ERROR yamux::connection] 2c3f5ad3: socket error: decode error: i/o error: I/O error: Connection reset by peer (os error 104)
Jun 16 14:08:06.000 INFO Syncing                                 est_time: 1 days 4 hrs, speed: 1.31 slots/sec, distance: 135546 slots (2 weeks 4 days), peers: 11, service: slot_notifier
Jun 16 14:08:18.000 INFO Syncing                                 est_time: --, speed: 0.00 slots/sec, distance: 135547 slots (2 weeks 4 days), peers: 12, service: slot_notifier
Jun 16 14:08:30.000 INFO Syncing                                 est_time: --, speed: 0.00 slots/sec, distance: 135548 slots (2 weeks 4 days), peers: 12, service: slot_notifier
Jun 16 14:08:42.000 INFO Syncing                                 est_time: --, speed: 0.00 slots/sec, distance: 135549 slots (2 weeks 4 days), peers: 12, service: slot_notifier
```

## Expected Behaviour

Sync Witti

## Steps to resolve

N/A

**Comment by leckylao** [comment]
I had similar issue today, beacon node got stopped with similar error. Restarted beacon node it sync passed those blocks then back to normal

![image](https://user-images.githubusercontent.com/148800/84778028-b8dd2280-b025-11ea-8850-583a5af80040.png)

![image](https://user-images.githubusercontent.com/148800/84778045-c09cc700-b025-11ea-8bfc-3311b330085a.png)

**Comment by AgeManning** [comment]
These are unrelated issues.

@paulhauner - The second seems to be a deadlock on the cache. 

@q9f - This is a sync bug - Likely you don't have debug logs? These kinds of errors can happen on edge case forks, but are hard to diagnose with just INFO logs.

**Comment by paulhauner** [comment]
@leckylao thanks for the report! I have two questions, if you will?

- What CPU is this running on. Rough description is fine, e.g. a Raspberry Pi or a 16 core beast.
- How many times did you see this error message? I'm interested in knowing if this happened a few times intermittently or if it happened without fail for several minutes.

**Comment by leckylao** [comment]
> @leckylao thanks for the report! I have two questions, if you will?
> 
> * What CPU is this running on. Rough description is fine, e.g. a Raspberry Pi or a 16 core beast.

It runs on single core cpu. 

> * How many times did you see this error message? I'm interested in knowing if this happened a few times intermittently or if it happened without fail for several minutes.

First time so far. After restart now has been running fine for a day now. Will keep an eye on it.

**Comment by AgeManning** [comment]
We've isolated the sync bug and should have that sorted. Thanks for the report @q9f

**Comment by paulhauner** [comment]
Copy that, thanks @leckylao. Given it's running on a single core and we have the discv5 CPU usage issue (fixed by #1274) I'm confident to put this down to resource contention causing the lock to be held for more than the timeout, instead of a logic bug that we need to deal with explicitly.

**Comment by ChuckNorrison** [comment]
iam at this point with the lighthouse 0.2.9... purge-db?
